src/pages/commits/ProjectSelect.tsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import IconButton from "@leafygreen-ui/icon-button";
+import { uiColors } from "@leafygreen-ui/palette";
+import { Body } from "@leafygreen-ui/typography";
+import { Input } from "antd";
+import Icon from "components/icons/Icon";
+import {
+  GetProjectsQuery,
+  GetProjectsQueryVariables,
+  ProjectFragment,
+} from "gql/generated/types";
+import { GET_PROJECTS } from "gql/queries";
+import { FavoriteStar } from "./projectSelect/FavoriteStar";
+import { ProjectOptionGroup } from "./projectSelect/ProjectOptionGroup";
+
+const { Search } = Input;
+const { gray, white } = uiColors;
+
+const filterProjects = (
+  projects: ProjectFragment[],
+  search: string,
+  favoriteIdentifiers: string[]
+) => {
+  if (search !== "") {
+    return projects.filter((project) => project.displayName.includes(search));
+  }
+
+  return projects.filter(
+    (project) => !favoriteIdentifiers.includes(project.identifier)
```

Comment:
this presents a difficult ux since `.includes` case sensitive. can we use regex to make a looser match?